expr/literals.go

```diff
@@ -406,21 +463,68 @@ type ProtoLiteral struct {
 	Type  types.Type
 }
 
+func timeFromPrecisionUnits(units int64, precision types.TimePrecision) time.Time {
+	var tm time.Time
+	switch precision {
+	case types.PrecisionSeconds:
+		tm = time.Unix(units, 0)
+	case types.PrecisionDeciSeconds:
+		tm = time.Unix(units/10, units%10*100000000)
+	case types.PrecisionCentiSeconds:
+		tm = time.Unix(units/100, units%100*10000000)
+	case types.PrecisionMilliSeconds:
+		tm = time.UnixMilli(units)
+	case types.PrecisionEMinus4Seconds:
+		tm = time.Unix(units/10000, units%10000*100000)
+	case types.PrecisionEMinus5Seconds:
+		tm = time.Unix(units/100000, units%100000*10000)
+	case types.PrecisionMicroSeconds:
+		tm = time.UnixMicro(units)
+	case types.PrecisionEMinus7Seconds:
+		tm = time.Unix(units/10000000, units%10000000*100)
+	case types.PrecisionEMinus8Seconds:
+		tm = time.Unix(units/100000000, units%100000000*10)
+	case types.PrecisionNanoSeconds:
+		tm = time.Unix(units/1000000000, units%1000000000)
+	default:
+		panic("unsupported precision")
+	}
+	return tm
+}
+
 func (t *ProtoLiteral) ValueString() string {
 	switch literalType := t.Type.(type) {
-	case *types.PrecisionTimestampType, *types.PrecisionTimestampTzType:
-		return fmt.Sprintf("%d", t.Value)
+	case *types.PrecisionTimestampType:
+		x, _ := t.Value.(int64)
```

Comment:
If you're ignoring the second param here, why bother having this as a separate line? 

Just do `tm := timeFromPrecisionUnits(t.Value.(int64), literalType.Precision)`?

Reply:
done